gapis/api/vulkan/transform_command_disabler.go

```diff
@@ -251,25 +250,48 @@ func (disablerTransform *commandDisabler) rewriteCommandBuffer(ctx context.Conte
 		currentCmd := existingCommandBuffer.CommandReferences().Get(uint32(i))
 		args := GetCommandArgs(ctx, currentCmd, st)
 
-		if !disablerTransform.shouldBeDisabled(currentSubCmdID) {
-			cleanup, newCmd, err := AddCommand(ctx, cb, newCmdBuffer, inputState, inputState, args)
+		if disablerTransform.shouldBeDisabled(currentSubCmdID) {
+			if !isCmdAllowedToDisable(args) {
+				return VkCommandBuffer(0), fmt.Errorf("Command type is not allowed to be disabled : %v", args)
+			}
+
+			// Skip the disabled command and do not copy it to the new command buffer
+			disablerTransform.removeFromDisabledList(currentSubCmdID)
+			log.I(ctx, "Command %v disabled", currentSubCmdID)
+			continue
+		}
+
+		if disablerTransform.doesContainDisabledCmd(currentSubCmdID) {
+			// Rewrite the secondary command buffer and create a new VkCmdExecuteCommands
+			// to use new command buffer instead the old one
+			executeCommandsArgs, ok := args.(VkCmdExecuteCommandsArgsʳ)
+			if !ok {
+				return VkCommandBuffer(0), fmt.Errorf("VkExecuteCommands could not found %v: ", args)
+			}
+
+			newCmd, err := disablerTransform.rewriteExecuteSecondaryCommandBuffer(ctx,
+				currentSubCmdID, newCmdBuffer, executeCommandsArgs, cmd, inputState)
 			if err != nil {
-				panic(fmt.Errorf("Invalid command-buffer detected %+v", err))
+				return VkCommandBuffer(0), fmt.Errorf("Error during rewriting secondary command buffer")
 			}
 			if err := disablerTransform.observeAndWriteCommand(newCmd); err != nil {
 				log.E(ctx, "Failed during adding command : [%v]%v", newCmd, err)
 				return VkCommandBuffer(0), err
 			}
-			cleanup()
+			log.I(ctx, "Secondary Command Buffer %v updated", currentSubCmdID)
 			continue
 		}
 
-		if !isCmdAllowedToDisable(args) {
-			return VkCommandBuffer(0), fmt.Errorf("Command type is not allowed to be disabled : %v", args)
+		// Copy the unaffected commands to the new command buffer
+		cleanup, newCmd, err := AddCommand(ctx, cb, newCmdBuffer, inputState, inputState, args)
+		if err != nil {
+			panic(fmt.Errorf("Invalid command-buffer detected %+v", err))
```

Comment:
This panic message mentions an invalid command buffer, but should it rather say "Cannot add command: %+v" ?

Reply:
Makes sense.